http-clients/aws-crt-client/src/test/java/software/amazon/awssdk/http/crt/AwsCrtHttpClientWireMockTest.java

```diff
@@ -0,0 +1,107 @@
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package software.amazon.awssdk.http.crt;
+
+import static com.github.tomakehurst.wiremock.client.WireMock.aResponse;
+import static com.github.tomakehurst.wiremock.client.WireMock.any;
+import static com.github.tomakehurst.wiremock.client.WireMock.stubFor;
+import static com.github.tomakehurst.wiremock.client.WireMock.urlPathEqualTo;
+import static com.github.tomakehurst.wiremock.core.WireMockConfiguration.wireMockConfig;
+import static org.apache.commons.lang3.RandomStringUtils.randomAlphabetic;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+import static software.amazon.awssdk.http.HttpTestUtils.createProvider;
+import static software.amazon.awssdk.http.crt.CrtHttpClientTestUtils.createRequest;
+
+import com.github.tomakehurst.wiremock.junit.WireMockRule;
+import java.net.URI;
+import java.util.concurrent.TimeUnit;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.BeforeClass;
+import org.junit.Rule;
+import org.junit.Test;
+import software.amazon.awssdk.crt.CrtResource;
+import software.amazon.awssdk.crt.io.EventLoopGroup;
+import software.amazon.awssdk.crt.io.HostResolver;
+import software.amazon.awssdk.http.RecordingNetworkTrafficListener;
+import software.amazon.awssdk.http.RecordingResponseHandler;
+import software.amazon.awssdk.http.SdkHttpRequest;
+import software.amazon.awssdk.http.async.AsyncExecuteRequest;
+import software.amazon.awssdk.http.async.SdkAsyncHttpClient;
+import software.amazon.awssdk.utils.Logger;
+
+public class AwsCrtHttpClientWireMockTest {
+    private static final Logger log = Logger.loggerFor(AwsCrtHttpClientWireMockTest.class);
+    private final RecordingNetworkTrafficListener wiremockTrafficListener = new RecordingNetworkTrafficListener();
+
+    @Rule
+    public WireMockRule mockServer = new WireMockRule(wireMockConfig()
+                                                          .dynamicPort()
+                                                          .dynamicHttpsPort()
+                                                          .networkTrafficListener(wiremockTrafficListener));
+
+    @BeforeClass
+    public static void setup() {
+        System.setProperty("aws.crt.debugnative", "true");
+    }
+
+    @Before
+    public void methodSetup() {
+        wiremockTrafficListener.reset();
+    }
+
+    @After
+    public void tearDown() {
+        // Verify there is no resource leak.
+        EventLoopGroup.closeStaticDefault();
+        HostResolver.closeStaticDefault();
+        CrtResource.waitForNoResources();
+    }
+
+    @Test
+    public void closeClient_reuse_throwException() throws Exception {
+        SdkAsyncHttpClient client = AwsCrtAsyncHttpClient.create();
+
+        client.close();
+        assertThatThrownBy(() -> makeSimpleRequest(client)).hasMessageContaining("is closed");
+    }
+
+    @Test
```

Comment:
Do we need a similar one for the host resolver?

Reply:
The test name might be a bit confusing, but it should cover the shared host resolver because they are sharing a static host resolver.

Reply:
Gotcha, yeah I was thrown off by the name.